### PR TITLE
[chip-tool] Add multiple writes in a single transaction support

### LINKED ISSUE #19139: [TC-IDM-3.1]Chip-tool support needed to send a WriteRequest message from TH to DUT to write any attribute on two clusters.
#### Problem
As per Test-Plan, test-step 15 says "DUT sends a WriteRequestMessage to the DUT to both the clusters with the appropriate dataversions(received in the previous step) to modify the value of an attribute ",
Chip-tool support needed to send a  WriteRequestMessage  from DUT  to TH to read any attribute on two clusters.

**PFA log below:-**
image

Spec Reference:-
https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/data_model/Interaction-Model.adoc#ref_WriteRequestAction

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

`chip-tool` does not supports multiple writes in a single interaction.

#### Change overview
 * Add supports to multiple writes in a single interaction.
    * `./out/debug/standalone/chip-tool onoff write on-time 13 0x12344321 1,2,3 `
    * `./out/debug/standalone/chip-tool onoff write-by-id 0x4001, 0x4001 '19',17' 0x12344321 1`
    * `./out/debug/standalone/chip-tool any write-by-id 0x6,0x7 0x4001,0x0 '5','false' 0x12344321 1,1`

fixes #19139

### CODE REVIEW

**Comment by vivien-apple** [comment]
This PR depends on #19575 since the `tv-casting-app` is on my way.

**Review by woody-apple — APPROVED** [review]
Fast tracking tooling updates

File: examples/chip-tool/commands/clusters/WriteAttributeCommand.h [hunk 1]
```diff
@@ -23,36 +23,105 @@
...
+        ModelCommand("write", credsIssuerConfig), mClusterIds(1, clusterId), mAttributeIds(1, attributeId)
+    {
+        AddArgument("attribute-name", attributeName, "The attribute name to write.");
+        AddArgument("attribute-value", static_cast<int64_t>(minValue), static_cast<uint64_t>(maxValue), &mAttributeValues,
+                    "The attribute value to write.");
+        AddArguments();
+    }
+
+    template <typename maxType>
+    WriteAttribute(chip::ClusterId clusterId, const char * attributeName, int minValue, maxType maxValue,
```
[the remark this answers is not in the document]
Comment (vivien-apple) [reply examples/chip-tool/commands/clusters/WriteAttributeCommand.h]: I update the template to `template <typename minType, typename maxType>`. For some reasons it was not working in the early version of this patch and I was lazy to investigate the different templates on `Command.h`.
Thanks for fixing my laziness!

File: examples/chip-tool/commands/common/Command.cpp [hunk 2]
```diff
@@ -305,6 +305,23 @@ bool Command::InitArgument(size_t argIndex, char * argValue)
...
+    case ArgumentType::VectorCustom: {
+        auto vectorArgument = static_cast<std::vector<CustomArgument *> *>(arg.value);
+
+        std::stringstream ss(argValue);
+        while (ss.good())
+        {
+            std::string valueAsString;
+            getline(ss, valueAsString, ',');
+
+            CustomArgument * customArgument = new CustomArgument();
```
[the remark this answers is not in the document]
Comment (vivien-apple) [reply examples/chip-tool/commands/common/Command.cpp:317]: They are here: https://github.com/project-chip/connectedhomeip/pull/19566/files#diff-a9e9c28744dcf9b0e0e95e36634b31e1b42d5dfd05580b811e9227226bc52f29R892

**Comment by vivien-apple** [comment]
I have included https://github.com/project-chip/connectedhomeip/pull/19575 in this PR. This is not in order to land it with it, but just to get the code to compile and see if CI is happy.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**